File: PizzaCommercial.cpp
#include <iostream>
#include <ctime>

using namespace std;

int main()
{   
    //clock_t timereq;
    //timereq = clock();
    long n;
    long p;
    cin>> n >>p;
    long m1=0,m2=0;
    int foo[n];
    for(int i=0;i<n;i++){
        cin>>foo[i];
        foo[i] = foo[i]-p; //Profit is calculated here
        m1 = foo[i] > foo[i] + m1 ? foo[i] : foo[i]+m1; //comparing the profit with previous profit   
        m2 = m1> m2 ? m1 : m2; //max profit so far is stored in m2
        //cout<<foo[i]<<endl;    
    }
    

    //timereq = clock() - timereq;
    cout<<m2<<endl;
    //cout<<(float)timereq/CLOCKS_PER_SEC<<endl;
   return 0;
}

File: PizzaCommercial.java
import java.util.Scanner;
    
public class RadioCommercial {
    int n,p;
    Scanner scan = new Scanner(System.in);
    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);

        RadioCommercial rc = new RadioCommercial();
        System.out.println(rc.maxProfits());
    }
    public int maxProfits(){
        n = scan.nextInt();
        p = scan.nextInt();
        int m1=0;
        int m2=0;
        int[] listening = new int[n];
        for(int i=0;i<n;i++){
            listening[i] = scan.nextInt();
            listening[i] = listening[i]-p; //Profit is calculated here
            m1 = listening[i] > listening[i] + m1 ? listening[i] : listening[i]+m1; //comparing the profit with previous profit   
            m2 = m1> m2 ? m1 : m2; //max profit so far is stored in m2
             
            }
        return m2;
    }
}

